src/map/lua/lua_baseentity.cpp

```diff
@@ -17709,6 +17710,98 @@ uint8 CLuaBaseEntity::getMannequinPose(uint16 itemID)
     return 0;
 }
 
+// Fishing Contest Utilities
+
+void CLuaBaseEntity::submitContestFish(uint32 score)
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        fishingcontest::SubmitFish(PChar, score);
+    }
+}
+
+void CLuaBaseEntity::withdrawContestFish()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        fishingcontest::WithdrawFish(PChar);
+    }
+}
+
+uint32 CLuaBaseEntity::getContestScore()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (FishingContestEntry* entry = fishingcontest::GetPlayerEntry(PChar))
+        {
+            return entry->score;
+        }
+    }
+
+    return 0;
+}
+
+uint8 CLuaBaseEntity::getContestRank()
+{
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (FishingContestEntry* entry = fishingcontest::GetPlayerEntry(PChar))
+        {
+            return entry->contestRank;
+        }
+    }
+
+    return 0;
+}
+
+auto CLuaBaseEntity::getContestRewardStatus() -> sol::table
+{
+    auto reward = lua.create_table();
+
+    if (m_PBaseEntity->objtype == TYPE_PC)
+    {
+        auto*       PChar = static_cast<CCharEntity*>(m_PBaseEntity);
+        const char* Query = "SELECT contestrank, share "
+                            "FROM   fishing_contest_entries "
+                            "WHERE  name = '%s' "
+                            "AND    claimed != 1; ";
+
+        int32 ret = _sql->Query(Query, PChar->getName());
```

Comment:
Please use the new `db::` system with `preparedStatement()`, there's lots of examples in `xi_world` and `xi_search`